johnpapa/vscode-peacock
# Feature/faves

# Favorites Take Center Stage

## Purpose


Features and Breaking Changes

- Added settings which allow the user to override the dark and light foreground colors
  - _darkForegroundColor_ setting (defaults to `15202b`) which can override the dark foreground on what is calculated as a light background
  - _lightForegroundColor_ setting (defaults to `e7e7e7`) which can override the light foreground on what is calculated as a dark background
- `Add recommended favorites` is now a command
- Recommended favorites are now being saved when a new version is installed. Avoid using the same names.
  - Recommended favorites are a list of constants found in `favorites.ts`. These are alphabetized.
  - Recommended favorites are a starting point for favorites. They will be installed whenever a new version is installed. They will extend your existing favorites, so feel free to continue to add to your local favorites! However be careful not to change the color of the recommended favorites as they will be overridden when a new version is installed.
  - This list may change from version to version depending on the Peacock authoring team.
- Built-in colors for Angular, React and Vue have been removed in favor of "favorites"
- One Built-in color exists: Peacock Green. This gives a user an easy color choice :)

## How to Test

`npm run test`

save to faves:

- run the extension (debug or uninstall, package to a vsix, and load)
- validate that recommended faves are saved to your user settings
- validate that a union is performed for these faves
- validate that if you change a value for a recommended fave, that it is overridden
- validate that the settings are not written unless a new version of Peacock appears

## Checklist of changes included

- [ x] CHANGELOG updated
- [ x] README updated
- [ x] Tests updated/added
- [ x] Prettier formatting
- [ x] console.log removed
- [ x] Dead code removed
- [ x] Unnecessary comments removed
- [ x] Images updated (if applicable)



@musicfuel I pushed v2 up to this PR and request you review. Would like your feedback specifically on two items:

1. should we reduce the starting set of favorites?
2. should we revise the logic to re-save the favorites when a new version is installed/updated?


addresses #112 and #109
